core/src/test/java/org/elasticsearch/action/support/master/TransportMasterNodeActionTests.java

```diff
@@ -320,4 +322,35 @@ protected void processBeforeDelegationToMaster(Request request, ClusterState sta
             }
         }
     }
+
+    public void testMasterFailoverAfterStepDown() throws ExecutionException, InterruptedException {
+        Request request = new Request();
+        PlainActionFuture<Response> listener = new PlainActionFuture<>();
+
+        final Response response = new Response();
+
+        clusterService.setState(ClusterStateCreationUtils.state(localNode, localNode, allNodes));
+
+        new Action(Settings.EMPTY, "testAction", transportService, clusterService, threadPool) {
```

Comment:
Maybe I misunderstood something here, but isn't that done by the first line in the method?
```
Request request = new Request().masterNodeTimeout(TimeValue.timeValueHours(1));
```